backend/dataall/modules/s3_datasets/api/dataset/types.py

```diff
@@ -23,6 +24,22 @@
     ],
 )
 
+DatasetRestrictedInformation = gql.ObjectType(
+    name='DatasetRestrictedInformation',
+    fields=[
+        gql.Field(name='AwsAccountId', type=gql.String),
+        gql.Field(name='region', type=gql.String),
+        gql.Field(name='S3BucketName', type=gql.String),
+        gql.Field(name='GlueDatabaseName', type=gql.String),
```

Comment:
nit: A call out that these 2 fields (`S3BucketName` and `GlueDatabaseName`) are a part of index in opensearch (`backend/dataall/modules/s3_datasets/indexers/dataset_indexer.py`)  - not sure really how "restricted" they are treated if that information can be found elsewhere